server/data/data-jmap/src/main/java/org/apache/james/jmap/api/projections/MessageFastViewPrecomputedProperties.java

```diff
@@ -66,6 +73,27 @@ public MessageFastViewPrecomputedProperties build() {
         return preview -> hasAttachment -> new Builder.FinalStage(preview, hasAttachment);
     }
 
+    public static class Factory {
+        private final Preview.Factory previewFactory;
+
+        @Inject
+        public Factory(Preview.Factory previewFactory) {
+            this.previewFactory = previewFactory;
+        }
+
+        public MessageFastViewPrecomputedProperties from(MessageResult messageResult) throws MailboxException, IOException {
+            return builder()
+                .preview(previewFactory.fromMessageResult(messageResult))
+                .hasAttachment(hasAttachment(messageResult.getLoadedAttachments()))
+                .build();
+        }
+
+        private boolean hasAttachment(List<MessageAttachment> attachments) {
```

Comment:
maybe `hasNotInlineAttachment` ? or a better worded version of that ?

Reply:
I don't like negative wording.

I would prefer `hasAttachmentWithContentDispositionAttachment` if we want to improve this point...